src/index.ts

```diff
@@ -1,130 +0,0 @@
-import {configLogs, error, getErrorMessage, info} from "./utils/logger";
-import {Config, loadConfig} from "./utils/configFileReader";
-import {PullRequestInfo} from "./pullRequestInfo";
-import {PipelineRun} from "./utils/piplineRun";
-import {Deployment, KubernetesAPI} from "./utils/kubernetes";
-
-const args = process.argv.slice(2);
-const configFileIndex = args.findIndex(arg => arg === '-p');
-let config: Config | undefined;
```

Comment:
Please eliminate this `let`, so that you don't have to use `config!` everywhere. Just move lines 15–22 into a function that returns or throws.